packages/candy_editor/src/assets/index.tsx

```diff
@@ -103,71 +119,143 @@ export const CandyDataEditor = () => {
   useEffect(() => {
     setCandyType(NOT_SELECTED);
     closeModal();
+    console.log('CANDYCLASS', candyClass);
   }, [editableCandyClass]);
 
   useEffect(() => {
     console.log('🍬 CANDYCLASS', candyClass);
   }, [candyClass]);
 
+  useEffect(() => {
+    if (candyDataEditor.existingCandyClass) {
+      setCandyClass(candyDataEditor.existingCandyClass);
+      let arrayPropertyWithId: PropertyWithId[] = [];
+      candyDataEditor.existingCandyClass.Class.forEach((property) => {
+        const propertyWithId: PropertyWithId = {
+          name: property.name,
+          immutable: property.immutable,
+          value: property.value,
+          id: Math.random().toString(),
+        };
+        arrayPropertyWithId.push(propertyWithId);
+      });
+      setEditableCandyClass({ Class: arrayPropertyWithId });
+    }
+
+    if (candyDataEditor.readOnly) {
+      setReadOnly(true);
+    }
+  }, []);
+
   return (
     <Card type="outlined" padding="16px">
       <Container>
         <Flex flexFlow="column" gap={16}>
           <Flex>
-            <Button size="small" btnType="filled" onClick={displayModal}>
-              + Add Candy Value
-            </Button>
-          </Flex>
-          {editableCandyClass.Class.length > 0 ? (
-            <>
-              <HR marginTop={8} marginBottom={8} />
-              <Grid columns={4} gap={16}>
-                <Grid column={1}>Property Name</Grid>
-                <Grid column={2}>Property Value</Grid>
-                <Grid column={3}>Immutable</Grid>
-                <Grid column={4}>Actions</Grid>
-              </Grid>
+            {candyDataEditor.readOnly ? (
```

Comment:
You don't need a ternary here since both cases are almost identical. Refactor suggestion:
```
              <>
                <Toggle
                  checked={readOnly}
                  disabled={candyDataEditor.readOnly}
                  handleToggle={() => handleToggleReadOnly()}
                  style={{ marginBottom: 'auto', marginTop: 'auto' }}
                />
                <span style={{ marginLeft: '16px' }}>
                   { candyDataEditor.readOnly ? readOnlyMessage : writableMessage}
                </span>
              </>
```